File: leetcode.com/binary-search.py
#!/bin/python3
# https://leetcode.com/problems/binary-search/

class Solution:
    def search(self, nums: list[int], target: int) -> int:
        l = 0
        r = len(nums) - 1

        while l <= r:
            middle = l + ((r - l) // 2)

            if nums[middle] == target:
                return middle
            elif nums[middle] < target:
                l = middle + 1
            else:
                r = middle -1

        return -1

File: leetcode.com/binary-search.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer}
def search(nums, target)
    left = 0
    right = nums.length - 1

    while left <= right do
        middle = (left + right) / 2

        if nums[middle] == target then
            return middle
        elsif nums[middle] > target then 
            right = middle - 1
        else
            left = middle + 1
        end
    end

    return -1
end s